### PR TITLE
feat(cmf): parse setting to apply i18n

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**

lots of labels are in the cmf settings.
Those settings can't be translate

**What is the chosen solution to this problem?**

make cmf-settings able to:
* parse settings to apply i18n
* parse settings to extract i18n

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/scripts/cmf-settings.i18n.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+/**
+ * getPathFromPattern - get the path from the current pattern
+ *
+ * @param  {string} pattern   is the template string of the path with {{namespace}} and {{local}}
+ * @param  {string} namespace namespace to set
+ * @param  {string} locale    local to set
+ * @return {string}           path with replaced value
+ */
+function getPathFromPattern(pattern, namespace, locale) {
+	const replaceMap = {
```
Comment (jmfrancois) [inline packages/cmf/scripts/cmf-settings.i18n.js]: please check if pattenr contains {{namespace}} and {{local}} using the same regex (moved in a constant)

File: packages/cmf/scripts/cmf-settings.i18n.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+ *
+ * @param  {object} settings  JSON Object
+ * @param  {string} namespace namespace to parse
+ * @return {Map}              dictionary of key/value locales
+ */
+function getLocaleByNamespace(settings, namespace) {
+	return jsonpath
+		.query(settings, JSON_PATH_EXPRESSION)
+		.reduce(
+			(locale, i18n) => locale.set(i18n.key.split(`${namespace}:`)[1], i18n.options.defaultValue),
```
Comment (jmfrancois) [inline packages/cmf/scripts/cmf-settings.i18n.js]: this namespace make our translation very repetitive. We should put it with a default value in cmf.json and put it in a extra attribute. (i18n.domain)

File: packages/cmf/scripts/cmf-settings.i18n.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+}
+
+/**
+ * getLocaleByNamespace - transform a JSON to a dictionary of key/value with a given namespace
+ *
+ * @param  {object} settings  JSON Object
+ * @param  {string} namespace namespace to parse
+ * @return {Map}              dictionary of key/value locales
+ */
+function getLocaleByNamespace(settings, namespace) {
```
Comment (jmfrancois) [inline packages/cmf/scripts/cmf-settings.i18n.js]: getLocal**sFrom**Namespace ?

File: packages/cmf/scripts/cmf-settings.i18n.js (hunk 2 above)
Comment (jare-talend) [reply packages/cmf/scripts/cmf-settings.i18n.js]: it's the syntax of i18n.t(key) where key is a string with a namespace and a key separated with a colon. e.g. namespace:key

File: packages/cmf/scripts/cmf-settings.i18n.js (hunk 3 above)
Comment (jare-talend) [reply packages/cmf/scripts/cmf-settings.i18n.js]: rather getLocalesFromNamespace.

File: packages/cmf/scripts/cmf-settings.merge.js [hunk 4]
```diff
@@ -1,144 +1,151 @@
...
 
-	function error(...args) {
-		logger ? logger(args) : console.error(args); // eslint-disable-line no-console
-		return onError();
-	}
+	// Init some stuff to use next
+	const cmfconfigPath = path.join(process.cwd(), DEFAULT_CONFIG_FILENAME);
+	const cmfconfig = importAndValidate(cmfconfigPath, onError);
+
+	let sources;
```
[the remark this answers is not in the document]
Comment (jare-talend) [reply packages/cmf/scripts/cmf-settings.merge.js]: It's a legacy code. we need to refactor all and it's not the purpose. I just cleaned the main code to remove the function in a separated file.

File: packages/cmf/scripts/cmf-settings.merge.js [hunk 5]
```diff
@@ -1,144 +1,151 @@
...
 	}
 
-	function overrideRoutes(path, settings) {
-		const childRoutes = getChildRoutes(path.split('/'), settings);
-		settings.overrideRoutes[path].forEach(config => {
-			childRoutes.push(config);
-		});
-		delete settings.overrideRoutes[path]; // eslint-disable-line no-param-reassign
+	// extract all keys from a folder
+	if (
```
[the remark this answers is not in the document]
Comment (jare-talend) [reply packages/cmf/scripts/cmf-settings.merge.js]: I don't want to touch a legacy code

**Pull request merged**